rust/examples/basic_operations.rs

```diff
@@ -0,0 +1,76 @@
+use arrow::{
+    array::{Int32Array, StringArray},
+    datatypes::{DataType, Field, Schema as ArrowSchema},
+    record_batch::RecordBatch,
+};
+use deltalake::operations::collect_sendable_stream;
+use deltalake::{action::SaveMode, DeltaOps, SchemaDataType, SchemaField};
+use std::sync::Arc;
+
+fn get_table_columns() -> Vec<SchemaField> {
+    vec![
+        SchemaField::new(
+            String::from("int"),
+            SchemaDataType::primitive(String::from("integer")),
+            false,
+            Default::default(),
+        ),
+        SchemaField::new(
+            String::from("string"),
+            SchemaDataType::primitive(String::from("string")),
+            true,
+            Default::default(),
+        ),
+    ]
+}
+
+fn get_table_batches() -> RecordBatch {
+    let schema = Arc::new(ArrowSchema::new(vec![
+        Field::new("int", DataType::Int32, false),
+        Field::new("string", DataType::Utf8, true),
+    ]));
+
+    let int_values = Int32Array::from(vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]);
+    let str_values = StringArray::from(vec!["A", "B", "A", "B", "A", "A", "A", "B", "B", "A", "A"]);
+
+    RecordBatch::try_new(schema, vec![Arc::new(int_values), Arc::new(str_values)]).unwrap()
+}
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() -> Result<(), deltalake::DeltaTableError> {
+    // create a delta operations client pointing at an un-initialized im memory location
+    // in a productive environment this would be created with "try_new" and point at
```

Comment:
```suggestion
    // Create a delta operations client pointing at an un-initialized in-memory location.
    // In a production environment this would be created with "try_new" and point at
```